webpack/JobInvocationDetail/JobInvocationHostTable.js

```diff
@@ -306,14 +306,37 @@ const JobInvocationHostTable = ({
 
   const { results = [] } = response;
 
-  const isHostExpanded = host => expandedHost.includes(host);
+  const isHostExpanded = host => expandedHost.includes(host.id);
+
   const setHostExpanded = (host, isExpanding = true) =>
     setExpandedHost(prevExpanded => {
-      const otherExpandedHosts = prevExpanded.filter(h => h !== host);
-      return isExpanding ? [...otherExpandedHosts, host] : otherExpandedHosts;
+      const otherExpandedHosts = prevExpanded.filter(h => h !== host.id);
+      return isExpanding
+        ? [...otherExpandedHosts, host.id]
+        : otherExpandedHosts;
     });
   const [showAlert, setShowAlert] = useState(false);
 
+  const pageHostIds = results.map(h => h.id);
+  const areAllPageRowsExpanded =
+    pageHostIds.length > 0 &&
+    pageHostIds.every(hostId => expandedHost.includes(hostId));
+
+  const onExpandAll = () => {
+    const allAreExpanded = areAllPageRowsExpanded;
+
+    setExpandedHost(prevExpanded => {
+      const pageIdsSet = new Set(pageHostIds);
+
+      if (allAreExpanded) {
```

Comment:
`const allAreExpanded = areAllPageRowsExpanded;`
Is this const needed? can the code use `areAllPageRowsExpanded`?